src/expressions/functions/builtInFunctions.sequences.ts

```diff
@@ -289,6 +290,26 @@ const fnUnordered: FunctionDefinitionType = (
 	return sequence;
 };
 
+const fnIndexOf: FunctionDefinitionType = (
+	dynamicContext,
+	_executionParameters,
+	_staticContext,
+	sequence,
+	search
+) => {
+	return search.mapAll(([onlySearchValue]) =>
+		sequence
+			.map((element, i) => {
```

Comment:
We miss atomization for the operands. See this example in the [specs](https://www.w3.org/TR/xpath-functions-31/#func-index-of): "The expression fn:index-of([1, [5, 6], [6, 7]], 6) returns (3, 4). (The array is atomized to a sequence of five integers)."

Regular [compares also do this](https://github.com/FontoXML/fontoxpath/blob/master/src/expressions/operators/compares/Compare.ts#L83) before calling `valueCompare`